pkg/sql/catalog/schematelemetry/scheduled_job_executor.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schematelemetry
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/schematelemetry/schematelemetrycontroller"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
+)
+
+type schemaTelemetryExecutor struct {
+	metrics schemaTelemetryMetrics
+}
+
+var _ jobs.ScheduledJobController = (*schemaTelemetryExecutor)(nil)
+var _ jobs.ScheduledJobExecutor = (*schemaTelemetryExecutor)(nil)
+
+type schemaTelemetryMetrics struct {
+	*jobs.ExecutorMetrics
+}
+
+var _ metric.Struct = &schemaTelemetryMetrics{}
+
+// MetricStruct implements metric.Struct interface.
+func (m *schemaTelemetryMetrics) MetricStruct() {}
+
+// OnDrop implements the jobs.ScheduledJobController interface.
+func (s schemaTelemetryExecutor) OnDrop(
+	ctx context.Context,
+	scheduleControllerEnv scheduledjobs.ScheduleControllerEnv,
+	env scheduledjobs.JobSchedulerEnv,
+	schedule *jobs.ScheduledJob,
+	txn *kv.Txn,
+	descsCol *descs.Collection,
+) error {
+	return nil
```

Comment:
https://github.com/cockroachdb/cockroach/blob/f9b7ec528bdbe22b12aea2afd3bc4c8cfc62d215/pkg/sql/compact_sql_stats.go#L169

Should we return an error to prevent dropping?

Reply:
Good catch. Yes, especially if we're going to remove that creation builtin.